### PR TITLE
Tests: Basic coverage with tree root domain

### PR DESCRIPTION
**Pull request by gkaihorodova**
Tests: Basic coverage with tree root domain
Extend existing legacy client tests to cover test cases with tree root domain.
https://fedorahosted.org/freeipa/ticket/6489

### CODE REVIEW

File: ipatests/test_integration/test_legacy_clients.py [hunk 1]
```diff
@@ -326,6 +328,91 @@ def test_login_disabled_subdomain_ad_user(self):
...
 
+    def test_getent_treedomain_ad_user(self):
+        if not self.ad_treedomain:
+            raise nose.SkipTest('AD tree root domain is not available.')
+
+        self.clear_sssd_caches()
+        testuser = 'treetestuser@{0}'.format(self.ad_treedomain)
+        result = self.legacy_client.run_command(['getent', 'passwd', testuser])
+
+        testuser_regex = "treetestuser@%s:\*:%s:%s:"\
```
Comment (martbab) [inline ipatests/test_integration/test_legacy_clients.py]: Backslashes in multiline strings are discouraged by PEP-8, please enclose the whole string literal in parentheses and leave out backslashes.

Also please try to use new-style string formatting in new code.

**Comment by martbab** [comment]
I have quickly skimmed through code and have one comment. Also, I have noticed the extreme code triplication of the test cases. I think that this warrants some refactoring first before adding tree-root domain tests.

File: ipatests/test_integration/test_legacy_clients.py (hunk 1 above)
Comment (gkaihorodova) [inline ipatests/test_integration/test_legacy_clients.py]: Thank you for review. I'll try to rewrite that part using new-style string formatting.

**Comment by gkaihorodova** [comment]
Can you be a little bit more specific about "triplication of the test cases ", please. 
Because, to be honest, I'm having hard time trying to navigate myself there.

**Comment by martbab** [comment]
If you look at the test cases (e.g. test_login_ipa-user, test_login_ad_user, test_login_subdomain_user are the 'best: examples) you can see that the function body is the same code copy-pasted with slight alterations so that it works for the new case. Your patch adds a *fourth* level of copy-pasta to the code, which is something that grieves me greatly.

Clearly, you can group the common code into a private method that can be only called with the use-case specific parameters for each test case. Or you can expand the existing mixing hierarchy to achieve this. Then it would also be simpler to extend the test cases for tree-root domains.

**Comment by gkaihorodova** [comment]
Thank you for explanation and tips. I noticed it as well and I agree that it (and not only that) worth refactoring. Yes, my PR is more or less copy-paste, because I was following existing pattern in the code.
Also I think PR can be pushed and at the same time feel free to open ticket/request or whatever is more suitable for refactoring  task and signed it to me. 
Please, don t be grieve, it makes me sad

**Comment by martbab** [comment]
Well you still have some issues to fix, notably the failing Travis CI and the not-so nice multiline-string literal.

**Comment by gkaihorodova** [comment]
Yes, sure I'll work on these issues

**Comment by gkaihorodova** [comment]
Bump for review

**Comment by martbab** [comment]
The patch looks ok, let's hope that our CI will play nice with it.

**Comment by gkaihorodova** [comment]
Thanks you for review. Let's hope for the best .

**Comment by martbab** [comment]
master:

* 10494b1bb34b6ff9c1b810cc0739c761b017202c Tests: Basic coverage with tree root domain

**Pull request closed without merging**